src/browser/fake_network.js

```diff
@@ -43,6 +58,220 @@ function iptolong(parts) {
+
+        /*
+        let head = this.head;
+        for(let i=0; i<src_length; ++i) {
+            buffer[head] = src_array[i];
+            head = (head + 1) % capacity;
+        }
+        this.head = head;
+        this.length += src_length;
+        */
+        const new_head = this.head + src_length;
+        if(new_head > capacity) {
+            const i_split = capacity - this.head;
+            buffer.set(src_array.subarray(0, i_split), this.head);
+            buffer.set(src_array.subarray(i_split));
+        }
+        else {
+            buffer.set(src_array, this.head);
+        }
+        this.head = new_head % capacity;
+        this.length += src_length;
+    }
+
+    /**
+     * @param {Uint8Array} dst_array
+     * @param {number} length
+     */
+    peek(dst_array, length)
+    {
+        if(length > this.length) {
+            length = this.length;
+        }
+        if(length) {
+            const buffer = this.buffer;
+            const capacity = buffer.length;
+            /*
+            for(let i=0, tail = this.tail; i<length; ++i) {
+                dst_array[i] = buffer[tail];
+                tail = (tail + 1) % capacity;
+            }
+            */
+            const new_tail = this.tail + length;
+            if(new_tail > capacity) {
+                const buf_len_left = new_tail % capacity;
+                const buf_len_right = capacity - this.tail;
+                dst_array.set(buffer.subarray(this.tail));
+                dst_array.set(buffer.subarray(0, buf_len_left), buf_len_right);
+            }
+            else {
+                dst_array.set(buffer.subarray(this.tail, new_tail));
+            }
+        }
+        return length;
+    }
+
+    /**
+     * @param {number} length
+     */
+    remove(length)
+    {
+        if(length > this.length) {
+            length = this.length;
+        }
+        if(length) {
+            this.tail = (this.tail + length) % this.buffer.length;
+            this.length -= length;
+        }
+        return length;
+    }
+}
+
+class EthernetPacketEncoder
+{
+    constructor()
+    {
+        const eth_frame = new Uint8Array(ETH_FRAME_SIZE);
+        const offset = eth_frame.byteOffset;
+
+        this.eth_frame = eth_frame;
+        this.eth_frame_view = new DataView(eth_frame.buffer);
+        this.eth_payload_view = new DataView(eth_frame.buffer, offset + ETH_PAYLOAD_OFFSET, ETH_PAYLOAD_SIZE);
+        this.ipv4_payload_view = new DataView(eth_frame.buffer, offset + IPV4_PAYLOAD_OFFSET, IPV4_PAYLOAD_SIZE);
+        this.udp_payload_view = new DataView(eth_frame.buffer, offset + UDP_PAYLOAD_OFFSET, UDP_PAYLOAD_SIZE);
+
+        for(const view of [this.eth_frame_view, this.eth_payload_view, this.ipv4_payload_view, this.udp_payload_view]) {
+            view.setArray = this.view_setArray.bind(this, view);
+            view.setString = this.view_setString.bind(this, view);
+            view.setInetChecksum = this.view_setInetChecksum.bind(this, view);
```

Comment:
`view_{setArray,setString,setInetChecksum}` should just be global functions, rather than doing this dance with monkeypatching built-in types.

Reply:
Fully agree, class EthernetPacketEncoder is gone now.